src/icon-manifest/iconGenerator.ts

```diff
@@ -0,0 +1,385 @@
+        const iconFileLightDefinition = `${sts.manifestFileLightPrefix}${icon}`;
+        const iconFileExtension = fileFormatToString(current.format);
+        const isFilename = current.filename;
+
+        defs[iconFileDefinition] = {
+          iconPath: filePath + suffix + iconFileExtension,
+        };
+
+        if (hasDefaultLightFile && !hasLightVersion) {
+          defs[iconFileLightDefinition] = {
+            iconPath: filePath + suffix + iconFileExtension,
+          };
+        }
+
+        if (hasLightVersion) {
+          defs[iconFileLightDefinition] = {
+            iconPath: fileLightPath + suffix + iconFileExtension,
+          };
+        }
+
+        if (current.languages) {
+          const assignLanguages = langId => {
+            languageIds[langId] = iconFileDefinition;
+          };
+
+          current.languages.forEach(langIds => {
+            if (Array.isArray(langIds.ids)) {
+              langIds.ids.forEach(id => { assignLanguages(id); });
+            } else {
+              assignLanguages(langIds.ids);
+            }
+          });
+        }
+
+        current.extensions.forEach(extension => {
+          if (isFilename) {
+            names.fileNames[extension] = iconFileDefinition;
+
+            if (hasDefaultLightFile && !hasLightVersion) {
+              light.fileNames[extension] = iconFileDefinition;
+            }
+
+            if (hasLightVersion) {
+              light.fileNames[extension] = iconFileLightDefinition;
+            }
+          } else {
+            names.fileExtensions[this.removeFirstDot(extension)] = iconFileDefinition;
+
+            if (hasDefaultLightFile && !hasLightVersion) {
+              light.fileExtensions[this.removeFirstDot(extension)] = iconFileDefinition;
+            }
+
+            if (hasLightVersion) {
+              light.fileExtensions[this.removeFirstDot(extension)] = iconFileLightDefinition;
+            }
+          }
+        });
+
+        return old;
+      }, {
+        defs: {},
+        names: { fileExtensions: {}, fileNames: {} },
+        light: { fileExtensions: {}, fileNames: {} },
+        languageIds: {},
+      });
+  }
+
+  private getRelativePath(fromDirPath, toDirName) {
+    if (toDirName == null) {
+      throw new Error('toDirName not defined.');
+    }
+
+    if (fromDirPath == null) {
+      throw new Error('fromDirPath not defined.');
+    }
+
+    if (!fs.existsSync(toDirName)) {
+      throw new Error('Directory \'' + toDirName + '\' not found.');
+    }
+    return './' +
+      path.relative(fromDirPath, toDirName).replace(/\\/g, '/') +
+      (toDirName.endsWith('/') ? '' : '/');
+  }
+
+  private removeFirstDot(txt: string): string {
+    if (txt.indexOf('.') === 0) {
+      return txt.substring(1, txt.length);
+    }
+    return txt;
+  }
+
+  private buildIconPath(
+    defaultExtension: IDefaultExtension,
+    schemaExtension: IIconPath,
+    path: string,
```

Comment:
Rename `path` to `iconsFolderBasePath` to not conflict with the name of the package.